homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,246 @@
+
+    try:
+        for device in flume_devices._devices:
+            if device["type"] == 2:
+                flume = FlumeData(
+                    username,
+                    password,
+                    client_id,
+                    client_secret,
+                    device["id"],
+                    time_zone,
+                    SCAN_INTERVAL,
+                )
+                sensor_name = name + " " + device["id"]
+                add_entities([FlumeSensor(flume, sensor_name)], True)
+    except Exception as error:
+        _LOGGER.error("Unable to setup Flume Devices: %s", error)
+        return False
+
+
+class FlumeSensor(Entity):
+    """Representation of the Flume sensor."""
+
+    def __init__(self, Flume, name):
+        """Initialize the Flume sensor."""
+        self.Flume = Flume
+        self._name = name
+        self._state = None
+        self._unit_of_measurement = "GALLONS"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the latest data and updates the states."""
+        self.Flume.update()
+        self._state = self.Flume.value
+
+
+class FlumeAuth:
+    """Get the Authentication Bearer, User ID and list of devices from Flume API."""
+
+    def __init__(self, username, password, client_id, client_secret):
+        """Initialize the data object."""
+        self._username = username
+        self._password = password
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._token = self.getToken()
+        self._user_id = self.getUserId()
+        self._bearer = self.getBearer()
+        self._devices = self.getDevices()
+
+    def getToken(self):
+        """Return authorization token for session."""
+        url = "https://api.flumetech.com/oauth/token"
+        payload = (
+            '{"grant_type":"password","client_id":"'
+            + self._client_id
+            + '","client_secret":"'
+            + self._client_secret
+            + '","username":"'
+            + self._username
+            + '","password":"'
+            + self._password
+            + '"}'
+        )
+        headers = {"content-type": "application/json"}
+
+        response = requests.request("POST", url, data=payload, headers=headers)
+
+        _LOGGER.debug("Token Payload: %s", payload)
+        _LOGGER.debug("Token Response: %s", response.text)
+
+        if response.status_code == 200:
+            return json.loads(response.text)["data"]
+        else:
+            raise Exception(
+                "getToken Response Code not Successful. Returned {}".format(
```

Comment:
Exception message should be updated for production.

Something like `Can't get token for user "{}". Response code returned : {}` (self._username, response.status_code)